src/compiler/checker.ts

```diff
@@ -12479,7 +12479,7 @@ namespace ts {
                 if (!node) return false;
                 switch (node.kind) {
                     case SyntaxKind.Identifier:
-                        return (node as Identifier).escapedText === argumentsSymbol.escapedName && getResolvedSymbol(node as Identifier) === argumentsSymbol;
+                        return (node as Identifier).escapedText === argumentsSymbol.escapedName && isExpressionNode(node) && getResolvedSymbol(node as Identifier) === argumentsSymbol;

```

Comment:
So this potentially fixes a broader set of issues. A more scoped fix would be to include `PropertyAssignment` below, but we might be missing other things which you can see if you take a look at `isIdentifierName`. Maybe we should be using `isIdentifierName`.

I'm honestly not sure; what are your thoughts?

Reply:
Oke, removed it.